### DISCUSSION TITLE
Prisma not using the correct Database

### DISCUSSION THREAD

**Discussion by Abh15h3k** [prisma/prisma#16931] [Q&A]
### Bug description

Following the Getting Started for Prisma with mongdb I've created a barebones project.

But when i run the code prisma is not able to connect to the database it gives the following error.

```
PrismaClientInitializationError: Database error. error code: unknown, error message: Server selection timeout: No available servers. Topology: { Type: ReplicaSetNoPrimary, Servers: [ { Address: mongo:27017, Type: Unknown, Error: failed to lookup address information: nodename nor servname provided, or not known }, ] }
    at startFn (/Users/abhishek/Documents/repo/prisma/node_modules/@prisma/client/runtime/index.js:27839:17)
    at async main (file:///Users/abhishek/Documents/repo/prisma/index.js:6:3) {
  clientVersion: '4.8.0',
  errorCode: undefined
}
```

it seems its trying to connect to `mongo:27017` but i've specified a different `DATABASE_URL` in the `.env.` file

i've also tried to export DATABASE_URL before running `npx prisma generate` and `node index.js`

### How to reproduce

Follow the Get Started for MongoDB.

Run the `node index.js`

### Expected behavior

It connects to the DB specified by DATABASE_URL

### Prisma information

```prisma
// Add your schema.prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User {
  id    String  @id @default(auto()) @map("_id") @db.ObjectId
  email String  @unique
  name  String?
}
```

```ts
// Add your code using Prisma Client
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

async function main() {
  console.log(proce)
  await prisma.$connect();
}

main()
  .then(async () => {
    const users = await prisma.user.findMany();
    console.log("Users", users);
  })
  .catch(async (e) => {
    console.error("DB connect error", e);
  })
  .finally(async () => {
    prisma.$disconnect();
  });

```

### Environment & setup

- OS: macOS (MacBook Pro M1 13 inch)
- Database: MongoDB
- Node.js version: 16.17.1
- MongoDB version: 4.2.10, running docker on a seperate system in the local network.

### Prisma Version

```
prisma                  : 4.8.0
@prisma/client          : 4.8.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.8.0-61.d6e67a83f971b175a593ccc12e15c4a757f93ffe
Default Engines Hash    : d6e67a83f971b175a593ccc12e15c4a757f93ffe
Studio                  : 0.479.0
```

**Comment by nurul3101** [comment]
Hey @Abh15h3k 👋 
Does your MongoDB Instance running locally have a replica set configured? 
If not, I would recommend you try a database hosted on Atlas, which has a replica set preconfigured. It is required to have a replica set configured to use the MongoDB connector.

I would also recommend you try connecting to MongoDB Instance through a Database IDE to confirm if it is accessible.

[MongoDB Docs Reference](https://www.prisma.io/docs/concepts/database-connectors/mongodb)

**Reply by Abh15h3k** [reply]
@nurul3101 
My MongoDB is running in docker which is on a different system but on the same network. I can access it through `mongosh` and `robo3t` by using the ip of that system. i have published port `27017` as well

It is also setup with replication.

I don't think where and how the mongodb is setup is the issue.

if you take a look at the error message it is not even trying to connect to the `DATABASE_URL` I have specified.
`{ Address: mongo:27017, Type: Unknown, Error: failed to lookup address information: nodename nor servname provided, or not known }`

in the `.env` file i have specified the following `DATABASE_URL="mongodb://hdocker:27017/primsa"`
`hdocker` is mapped to the ip of the system running mongodb using `/etc/hosts` file.

but when i run `node index.js`. the error always shows it is trying to connect to `mongo:27017`

**Reply by nurul3101** [reply]
Hmm, Can you try hardcoding the IP address in DATABASE_URL and then check if you are able to connect from the docker file?

**Comment by Ahmadre** [comment]
It is because Prisma appends the SAME Database url env in your .env file!!!! WHY?! Just replace it or leave it out!!!!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
